### ISSUE TITLE
Add a email address change workflow

### ISSUE THREAD

**Issue by mtrezza** [parse-community/parse-server#7152]
### New Feature / Enhancement Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Current Limitation
Currently there is not built-in workflow for changing an email address (or username) like there is for email verification or password reset.

### Feature / Enhancement Description
Adding a built-in workflow for changing the email address can leverage the already existing mechanisms for email verification / password reset.

### Example Use Case
1. User requests to change the email.
2. Parse Server sends email to old email address.
3. User opens email link and gets to website to enter a new email address.
4. Parse Server changes email address in DB and sends confirmation email to old and new email address.

### Alternatives / Workarounds
Manually implementing a workflow.

### 3rd Party References
(none)

**Comment by sommye-ctr** [comment]
Any updates on this?

**Comment by mtrezza** [comment]
Yes, the Pages router is capable of building this, but it is currently only available on the `master` branch and will be available from Parse Server 5.0.

But even with Parse Server 5.0 there won't be a built-in flow to change the email address, it still has to be built manually.

**Comment by andreisucman** [comment]
Hey @mtrezza. Can you roughly describe what needs to be done to build that flow manually? If I can do it I'll share it here.

**Comment by mtrezza** [comment]
@andreisucman

- For help with Parse Platform we recommend our [community forum](https://community.parseplatform.org) or our [community chat](https://chat.parseplatform.org)
- For general coding questions we recommend Stack Overflow using the [parse-platform](https://stackoverflow.com/questions/tagged/parse-platform) tag
